packages/block-editor/src/store/reducer.js

```diff
@@ -718,7 +718,20 @@ const BLOCK_SELECTION_INITIAL_STATE = {
 export function blockSelection( state = BLOCK_SELECTION_INITIAL_STATE, action ) {
 	switch ( action.type ) {
 		case 'CLEAR_SELECTED_BLOCK':
-			return BLOCK_SELECTION_INITIAL_STATE;
+			return {
+				...BLOCK_SELECTION_INITIAL_STATE,
+				previousSelection: {
+					start: state.start,
+					end: state.end,
+					isMultiSelecting: state.isMultiSelecting,
+					isEnabled: state.isEnabled,
+					initialPosition: state.initialPosition,
+				},
+			};
+		case 'RESTORE_SELECTED_BLOCK':
+			return {
+				...state.previousSelection,
```

Comment:
Do we need to spread? Or can we simply do:

```js
return state.previousSelection;
```

A spread operator is more costly, since it creates a new object, and performs a shallow clone of `previousSelection` into that new object.